api/classes/class-lightweight-api.php

```diff
@@ -224,23 +285,31 @@ public function get_client_data( $client_id, $do_not_rebuild = false ) {
 			$wpdb->prepare( "SELECT * FROM $events_table_name WHERE client_id = %s AND type = 'post_read'", $client_id ) // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 		);
 
+		// If client hasn't read any posts yet, don't write any data.
```

Comment:
This threw me off a bit, since the posts read are not the only data stored - it's possible to have a client with no posts read, but with a donation made. _But_ donations are not stored in the custom events table, and so do not matter here. I suggest a change in the comment:

```suggestion
		// If there is no relevant client data in the events table, do not save any data.
```